public/assets/js/blogPosts.js

```diff
@@ -0,0 +1,135 @@
+const createForm = $("#create-form");
+const updateForm = $("#update-form");
+const deletePost = $("#deletePost");
+const viewPost = $("#homePosts");
+const postComment = $("#postComment");
+
+const validate = (title, content) => {
+  // See if values contain text
+  newTitle = title.replace(/\s/g, "");
+  newContent = content.replace(/\s/g, "");
+
+  if (newTitle && newContent) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const handleCreate = async (event) => {
+  event.preventDefault();
+  const title = $("#title").val();
+  const content = $("#content").val();
+
+  const isValid = validate(title, content);
+
+  if (isValid) {
```

Comment:
Is there any way we can simplify and remove some of these nested `if/else` blocks? A good way to get rid of `else` at least is to have the `if` block check for the unhappy/error path. once we've caught all the possible errors, we can simply add the logic for the happy path in the main body of the function. this would look something like this:

```
if (!isValid) {
  renderError("blog-error", "Please fill in all fields");
}

const payload = { title, content };

    const response = await fetch("/api/blog/", {
      method: "POST",
      body: JSON.stringify(payload),
      headers: {
        "Content-Type": "application/json",
      },
    });

    const data = await response.json();

 if (!data.success) {
     renderError("blog-error", "Failed to create post. Try again.");
}

window.location.assign("/dashboard");


```